tests/ignite/metrics/regression/test_pearson_correlation.py

```diff
@@ -0,0 +1,224 @@
+from typing import Tuple
+
+import numpy as np
+import pytest
+import torch
+from torch import Tensor
+
+import ignite.distributed as idist
+from ignite.engine import Engine
+from ignite.exceptions import NotComputableError
+from ignite.metrics.regression import PearsonCorrelation
+
+
+def np_corr(np_y_pred: np.ndarray, np_y: np.ndarray, eps: float = 1e-8):
+    cov = np.cov(np_y_pred, np_y, ddof=0)[0, 1]
+    std_y_pred = np.std(np_y_pred, ddof=0)
+    std_y = np.std(np_y, ddof=0)
+    corr = cov / np.clip(std_y_pred * std_y, eps, None)
+    return corr
```

Comment:
@kzkadc by the way, can we use scipy pearsonr: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.pearsonr.html as a ref for the tests ?

Reply:
Thank you for the suggestion.
Unlike the scipy implementation, I added a small `eps` to the denomiantor to avoid zero division, which occurs when the variance of `y_pred` or `y` is zero, or `compute()` is called with only one sample.

I updated the test to use scipy ref for ordinary case and numpy ref for the above case.